### PR TITLE
SharpMap.Data.Providers.SqlServer2008 updates for Geography

### PR DESCRIPTION
**Pull request by terribletim**
With reference to Issue #83, fix the following in **SqlServer2008.cs**:
1. Use list of attribute columns in queries to fix problem with **g.*** affecting projects that do not reference Microsoft.SqlServer.Types (causes DataAdaptor to throw error

2. Use SqlServer **.ReorientObject()** when passing extent envelope / polygon
in queries for Geography data types, to ensure rings comply with SQL Server left-foot rule

3. Restrict bbox geometry for Geography types to -180, +180, -90, +90

4. Update `ExtentsMode ` with ArgOutOfRange exception for SpatialIndex if using Geography

5. Update `GetExtents` to use WKB (instead of WKT)

6. Update `GetExtents` to support Geography type on mode `QueryIndividualFeatures` using Geography::CollectionAggregate (closest equivalent to geometry::STEnvelope)

7. Update `GetExtents` to support Geography type on mode `EvenvelopeAggregate` using Geography::ConvexHullAggregate (closest equivalent to geometry::EnvelopeAggregate)

8. Add _logger to output SQL queries executed against database

And for project **Examples/WinFormsSamples**

1. Target .NetFramework 4.5 
2. Plug WebMercator spherical definition into CoordinateSystemServices so that it is available to vector layers setting up transformations by specifying Source and Target SRIDs
3. update BruTile from v.18 to v2.1
4. Add compressed LocalDb mdf with small geography and geometry tables
5. Add SqlServer (LocalDb) sample to FromMapBox with dialogue to interactively modify Data Provider properties (see image)

Regards

Tim

### CODE REVIEW

**Comment by terribletim** [comment]
Unit Tests updated for `SharpMap.Data.Providers.SqlServer2008` to support both geometry and geography types. Travis builds failed on some of the commits as I was attempting to use package **Microsoft.SqlServer.Types** but the Travis **nuget CLI** cannot run install scripts to copy files into project directory (requires `.targets` which isn't part of the package). So I have removed this package and updated reference to a local version of Microsoft.SqlServer.Types.dll<br/>

File: SharpMap/Data/Providers/SqlServer2008.cs [hunk 1]
```diff
@@ -86,7 +94,31 @@ public enum SqlServer2008ExtentsMode
...
         /// <summary>   
-        /// Initializes a new connection to SQL Server   
+        /// Initializes a new connection to SQL Server for Geometry spatial type in column named SHAPE
+        /// </summary>   
+        /// <param name="connectionStr">Connectionstring</param>   
+        /// <param name="tablename">Name of data table</param>   
+        /// <param name="oidColumnName">Name of column with unique identifier</param>   
+        public SqlServer2008(string connectionStr, string tablename, string oidColumnName)
+            : this(connectionStr, tablename, "shape", oidColumnName, SqlServerSpatialObjectType.Geometry)
+        {
```
Comment (FObermaier) [inline SharpMap/Data/Providers/SqlServer2008.cs:195]: "SHAPE"?

File: SharpMap/Data/Providers/SqlServer2008.cs [hunk 2]
```diff
@@ -86,7 +94,31 @@ public enum SqlServer2008ExtentsMode
...
+        /// Initializes a new connection to SQL Server for spatial column named SHAPE  
+        /// </summary>   
+        /// <param name="connectionStr">Connectionstring</param>   
+        /// <param name="tablename">Name of data table</param>   
+        /// <param name="oidColumnName">Name of column with unique identifier</param>
+        /// <param name="spatialObjectType">The type of the spatial object to use for spatial queries</param>
+        public SqlServer2008(string connectionStr, string tablename, string oidColumnName,
+            SqlServerSpatialObjectType spatialObjectType)
+            : this(connectionStr, tablename, "shape", oidColumnName, spatialObjectType)
+        {
```
Comment (FObermaier) [inline SharpMap/Data/Providers/SqlServer2008.cs:209]: SHAPE?

File: SharpMap/Data/Providers/SqlServer2008.cs [hunk 3]
```diff
@@ -86,7 +94,31 @@ public enum SqlServer2008ExtentsMode
...
+        }
+
+        /// <summary>   
+        /// Initializes a new connection to SQL Server for Geometry spatial type with default Extents mode
         /// </summary>
```
Comment (FObermaier) [inline SharpMap/Data/Providers/SqlServer2008.cs:214]: ... default `<see cref="ExtentsMode" />.`

**Pull request merged**